lib/internal/bootstrap/switches/does_own_process_state.js

```diff
@@ -138,7 +138,17 @@ function wrappedUmask(mask) {
 }
 
 function wrappedCwd() {
-  if (cachedCwd === '')
-    cachedCwd = rawMethods.cwd();
+  if (cachedCwd === '') {
+    try {
+      cachedCwd = rawMethods.cwd();
+    } catch (e) {
+      if (e.code === 'ENOENT' && e.syscall === 'uv_cwd') {
```

Comment:
I think @Qard also mentions this but by changing the error in the native code in `node_process_methods.cc` the error here will no longer have the `ENOENT` or `uv_cwd` in the error properties. If you're going to be changing the error thrown at the native layer why not simply add this additional bit to the message thrown at the native layer?

Reply:
Yeah, probably best to just do everything in the native code here. Probably don't need a JS-side wrapper error at all.